larana/OpticalDetector/SimPhotonCounter_module.cc

```diff
@@ -480,107 +491,96 @@ namespace opdet {
-          // add an empty record to the per event tree
-          if(fMakeOpDetEventsTree) fTheEventTree->Fill();
-        }
-        if(fMakeLightAnalysisTree) {
-          std::cout<<"Filling the analysis tree"<<std::endl;
-          //---------------Filling the analysis tree-----------:
-          fRun = evt.run();
-          std::vector<double> this_xyz;
-
-          //loop over the particles
-          for(size_t i_p=0; i_p < mcpartVec.size(); i_p++){
-
-	    const simb::MCParticle pPart = mcpartVec[i_p];
-	    if(pPart.Process() == "primary")
-	      fEnergy = pPart.E();
-
-            //resetting the vectors
-            fstepPositions.clear();
-            fstepTimes.clear();
-            fSignalsvuv.clear();
-            fSignalsvis.clear();
-            fdEdx = -1.;
-            //filling the tree fields
-	    fTrackID = pPart.TrackId();
-	    fpdg = pPart.PdgCode();
-	    fmotherTrackID = pPart.Mother();
-	    fdEdx = totalEnergy_track[fTrackID];
-	    fSignalsvuv = fSignals_vuv[fTrackID];
-	    fSignalsvis = fSignals_vis[fTrackID];
-	    fProcess = pPart.Process();
-            //filling the center positions of each step
-            for(size_t i_s=1; i_s < pPart.NumberTrajectoryPoints(); i_s++){
-	      this_xyz.clear();
-	      this_xyz.resize(3);
-	      this_xyz[0] = pPart.Position(i_s).X();
-	      this_xyz[1] = pPart.Position(i_s).Y();
-	      this_xyz[2] = pPart.Position(i_s).Z();
-	      fstepPositions.push_back(this_xyz);
-	      fstepTimes.push_back(pPart.Position(i_s).T());
-	    }
-            //filling the tree per track
-            fLightAnalysisTree->Fill();
           }
-	}
+          else
+          {
+            // if empty OpDet hit collection,
+            // add an empty record to the per event tree
+            if(fMakeOpDetEventsTree) fTheEventTree->Fill();
+          }
+          if(fMakeLightAnalysisTree) {
+            std::cout<<"Filling the analysis tree"<<std::endl;
+            //---------------Filling the analysis tree-----------:
+            fRun = evt.run();
+            std::vector<double> this_xyz;
+
+            //loop over the particles
+            for(simb::MCParticle const& pPart: *mcpartVec){
+
+              if(pPart.Process() == "primary")
+                fEnergy = pPart.E();
+
+              //resetting the vectors
+              fstepPositions.clear();
+              fstepTimes.clear();
+              fSignalsvuv.clear();
+              fSignalsvis.clear();
+              fdEdx = -1.;
+              //filling the tree fields
+              fTrackID = pPart.TrackId();
+              fpdg = pPart.PdgCode();
+              fmotherTrackID = pPart.Mother();
+              fdEdx = totalEnergy_track[fTrackID];
+              fSignalsvuv = fSignals_vuv[fTrackID];
+              fSignalsvis = fSignals_vis[fTrackID];
+              fProcess = pPart.Process();
+              //filling the center positions of each step
+              for(size_t i_s=1; i_s < pPart.NumberTrajectoryPoints(); i_s++){
```

Comment:
Can there ever be a situation where there are no trajectory points i.e. could this segfault?